modules/storages/app/common/storages/peripherals/storage_interaction/authentication_strategies/oauth_client_credentials.rb

```diff
@@ -37,41 +37,57 @@ def self.strategy
             Strategy.new(:oauth_client_credentials)
           end
 
-          def call(storage:, http_options: {}, &)
+          def call(storage:, http_options: {})
             config = storage.oauth_configuration.to_httpx_oauth_config
 
             return build_failure(storage) unless config.valid?
 
-            create_http_and_yield(issuer: config.issuer,
-                                  client_id: config.client_id,
-                                  client_secret: config.client_secret,
-                                  scope: config.scope,
-                                  http_options:,
-                                  &)
-          end
+            access_token = Rails.cache.read("storage.#{storage.id}.access_token")
```

Comment:
🟡 Could this be replaced by `cache.fetch`?

```ruby
access_token = Rails.cache.fetch('stinking.key') do
  handle_missing_token_and_return_it
end
```

Reply:
no, as the I cannot figure any block that returns only the token.

with httpx you create an http session, and while doing so, you get an access token attached to the session. Which is why I extract the token later to write the cache.